lib/vrl/diagnostic/src/note.rs

```diff
@@ -33,21 +37,40 @@ impl fmt::Display for Note {
         use Note::*;
 
         match self {
-            Hint(hint) => write!(f, "hint: {}", hint),
+            Hint(hint, url) => {
+                if url.is_empty() {
+                    write!(f, "hint: {}", hint)
+                } else {
+                    write!(f, "hint: {}\n    see: {}", hint, url)
```

Comment:
I don't think this will work/look good. The diagnostic messages have to be inserted line-by-line, since it adds some stuff in front of the line.

If you can't find a quick way to do so, I'd rather revert this.

Reply:
@JeanMertz Here's an example output:

```
error[E660]: non-boolean negation
  ┌─ :1:1
  │
1 │ !27
  │ ^-- this expression resolves to "integer"
  │ │
  │ negation only works on boolean values
  │
  = hint: coerce the value to the required type using a coercion function
        see: https://functions.vrl.dev/#coerce-functions
  = see language documentation at https://vrl.dev
```

Personally, I think it looks okay, but I'm not wedded to it so quite happy to leave it up to you.

Reply:
Okay, I've reworked this and used `SeeDocs` for the URL hint rather than `Hint`. I also added a utility struct for constructing URLs but I suspect you might find that to be overkill. If so, easy to revert.

Reply:
Here's how it looks now:

```
error[E660]: non-boolean negation
  ┌─ :1:1
  │
1 │ !27
  │ ^-- this expression resolves to "integer"
  │ │
  │ negation only works on boolean values
  │
  = hint: coerce the value to the required type using a coercion function
  = see type coercion documentation at https://functions.vrl.dev/#coerce-functions
  = see language documentation at https://vrl.dev
```